articles/node-rxjs/index.md

```diff
@@ -0,0 +1,146 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/node-rxjs/
+title: Node.js Rxjs
+description: This article shows developers how to use the RxJs library and some of its operators
+author: linus-muema
+date: 2020-09-08T00:00:00-14:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education//node-rxjs/hero.jpg
+    alt: node authentication image example key
+---
+ReactiveX stands for *Reactive Extensions*. It is a collection of projects done by the ReactiveX community. They bring asynchronous programming into different programming languages and platforms. Their observable patterns also helps in making the code smaller(less) and easier to read.
+They also have a well-managed error handling mechanism. When compared to the old `try/catch` method, it performs better. Some of these projects include [RxJava](https://github.com/ReactiveX/RxJava), [RxJs](https://github.com/ReactiveX/rxjs), [RxSwift](https://github.com/ReactiveX/RxSwift), etc.
+<!--more-->
+
+Companies like GitHub, Netflix, and Microsoft use these projects.
+
+Today we will look at how we can add RxJs to our Node.js applications using Typescript. We will also look at the various operators found in the library.
+
+To begin with, RxJs makes use of `observables` to observe and emit data. There are several ways to create observables. We can create them using the `create` function or use the `of`, `from`, `interval`, `range`, etc. operators. But, as from v.6+, `create` is deprecated and so we must create the Observable module.
+
+```typescript
```

Comment:
Your code looks pretty nice, but it looks like it's missing some stuff. Can I suggest writing this up in an online code sandbox, like [repl.it](repl.it). You can then drop the link to that demo in your article. This is the preferred way to just include code snippets while still linking to a running version of your tutorial.

Reply:
@LinusMuema This is probably the best way to get your article revised quickly in the future. I'm not as up to date with node as you are, but if I can see it work, then I know it's good to go. Let me know if  there's something that I'm missing here, just in case I misread something.

Reply:
Hi @nadivgold 
Thanks for the pointers. Can I fix the errors in the code and in the future articles I can then use replit. I am not sure about the broken image, is it the path that is wrong or there is a missing image?

Reply:
And I have a question about how you tested the code. Did you use the snippets as they are or you cloned the gothub repo?

Reply:
Aha I knew I missed something. Yeah just didn't see the repo. I was testing the snippets. Either way it's nice to have the snippets run-able in an online sandbox, but I'll take another look at the code.

Good catch.

Reply:
Okay. I have another question (I hope I'm not pestering you), can the replit snippets be runnable directly on the blog article. Like the way kotlin does on it's site.

[this way](https://kotlinlang.org/docs/reference/ranges.html)

Reply:
That would be cool. Unfortunately, Section doesn't want embedded code, as a security measure. For us, the best way to make run-able code is with a link to a dedicated code playground, at least for now. 

I will talk to the team @hectorkambow @ninjaginja, and I will see what I can do about embedding run-able code. For now, a link would be the best way to let readers interact with your code.

Reply:
Okay. I will add replit code snippets then. I hope the code at Github runs well